extensions/applicationinsights-clickanalytics-js/src/common/Utils.ts

```diff
@@ -0,0 +1,362 @@
+        if (attributeNamesExpectedObjects.hasOwnProperty(i)) {
+            var objectName = attributeNamesExpectedObjects[i];
+            if (typeof overrideConfig[objectName] === 'object') {
+                removeInvalidElements(overrideConfig[objectName]);
+            } else {
+                delete overrideConfig[objectName];
+            }
+        }
+    }
+}
+
+/**
+ * Finds attributes in object which are invalid 
+ * and deletes them. useful in override config
+ * @param object Input object
+ */
+export function removeInvalidElements(object: Object): void {
+    /// Because the config object 'callback' contains only functions, 
+    /// when it is stringified it returns the empty object. This explains
+    /// the workaround regarding 'callback'
+    for (var property in object) {
+        if (!isValueAssigned(object[property]) ||
+            (JSON.stringify(object[property]) === '{}' && (property !== 'callback'))) {
+            delete object[property];
+        }
+    }
+}
+
+/**
+ * Checks if value is assigned to the given param.
+ * @param value - The token from which the tenant id is to be extracted.
+ * @returns True/false denoting if value is assigned to the param.
+ */
+export function isValueAssigned(value: any) {
+    /// <summary> takes a value and checks for undefined, null and empty string </summary>
+    /// <param type="any"> value to be tested </param>
+    /// <returns> true if value is null undefined or emptyString </returns>
+    return !(CoreUtils.isNullOrUndefined(value) || value === '');
+}
+
+/**
+ * Determines whether an event is a right click or not 
+ * @param evt - Mouse event 
+ * @returns true if the event is a right click 
+ */
+export function isRightClick(evt: any): boolean {
+    if ('which' in evt) { // Chrome, FF, ...
+        return (evt.which === 3);
+    } else if ('button' in evt) { // IE, ...
+        return (evt.button === 2);
+    }
+}
+
+/**
+ * Determines whether an event is a left click or not 
+ * @param evt - Mouse event 
+ * @returns true if the event is a left click 
+ */
+export function isLeftClick(evt: any): boolean {
+    if ('which' in evt) { // Chrome, FF, ...
+        return (evt.which === 1);
+    } else if ('button' in evt) { // IE, ...
+        return (evt.button === 1);
+    }
+}
+
+/**
+ * Determines whether an event is a middle click or not 
+ * @param evt - Mouse event 
+ * @returns true if the event is a middle click 
+ */
+export function isMiddleClick(evt: any): boolean {
+    if ('which' in evt) { // Chrome, FF, ...
+        return (evt.which === 2);
+    } else if ('button' in evt) { // IE, ...
+        return (evt.button === 4);
+    }
+}
+
+/**
+ *  Determines whether an event is a keyboard enter or not
+ * @param evt - Keyboard event 
+ * @returns true if the event is a keyboard enter
+ */
+export function isKeyboardEnter(evt: KeyboardEvent): boolean {
+    if ('keyCode' in evt) { // Chrome, FF, ...
+        return (evt.keyCode === 13);
```

Comment:
nit: it would be nice to convert these magic numbers to const values at the top of the file while 13, 32 are recognizable (for anyone how has worked with ASCII) the above while === 2 and button === 4 causes me to go and look up what they mean rather than the code just being self explanatory. Feel free to create a separate bug / PR for this as I know it's just code that's been migrated.